rust/main/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -2,61 +2,50 @@
 #![allow(clippy::unnecessary_get_then_check)] // TODO: `rustc` 1.80.1 clippy issue
 
 use std::{
-    collections::HashMap,
     fmt::Debug,
-    ops::Deref,
-    str::FromStr,
     sync::Arc,
     time::{Duration, Instant},
 };
 
-use crate::{
-    merkle_tree::builder::MerkleTreeBuilder,
-    msg::metadata::{
-        multisig::{MerkleRootMultisigMetadataBuilder, MessageIdMultisigMetadataBuilder},
-        AggregationIsmMetadataBuilder, CcipReadIsmMetadataBuilder, NullMetadataBuilder,
-        RoutingIsmMetadataBuilder,
-    },
-    settings::matching_list::MatchingList,
-};
-use async_trait::async_trait;
+use crate::settings::matching_list::MatchingList;
 use derive_new::new;
-use eyre::{Context, Result};
-use hyperlane_base::{
-    db::{HyperlaneDb, HyperlaneRocksDB},
-    settings::CheckpointSyncerBuildError,
-};
-use hyperlane_base::{
-    settings::{ChainConf, CheckpointSyncerConf},
-    CheckpointSyncer, CoreMetrics, MultisigCheckpointSyncer,
-};
-use hyperlane_core::{
-    accumulator::merkle::Proof, AggregationIsm, CcipReadIsm, Checkpoint, HyperlaneDomain,
-    HyperlaneMessage, InterchainSecurityModule, Mailbox, ModuleType, MultisigIsm, RoutingIsm,
-    ValidatorAnnounce, H160, H256,
-};
+use eyre::Result;
+use hyperlane_core::{HyperlaneMessage, InterchainSecurityModule, Mailbox, ModuleType, H256};
 
 use tokio::sync::RwLock;
-use tracing::{debug, info, instrument, warn};
 
-#[derive(Debug, thiserror::Error)]
-pub enum MetadataBuilderError {
+#[derive(Clone, Debug, PartialEq, thiserror::Error)]
+pub enum MetadataBuildError {
+    /// While building metadata, encountered something that should
+    /// prohibit all metadata for the message from being built.
+    /// Provides the reason for the refusal.
+    #[error("Refused")]
+    Refused(String),
+    /// Unable to fetch metadata, but no error occurred
+    #[error("Could not fetch metadata")]
+    CouldNotFetch,
     #[error("Unknown or invalid module type ({0})")]
     UnsupportedModuleType(ModuleType),
     #[error("Exceeded max depth when building metadata ({0})")]
     MaxDepthExceeded(u32),
```

Comment:
Since we call new error `MaxIsmCountReached` (with `Ism` infix), I would add infix `Ism` into `MaxDepthExceeded` error as well.

Reply:
Ah good point!